frameworks/cassandra/tests/test_recovery.py

```diff
@@ -47,31 +47,28 @@ def test_node_replace_replaces_node():
     plan.wait_for_completed_recovery(PACKAGE_NAME)
 
     # get an exact task id to run 'task exec' against... just in case there's multiple cassandras
+    # Recovery will have completed after the line above so the task id will be stable.
     pod_statuses = json.loads(cmd.run_cli('cassandra pods status node-0'))
     task_id = [task['id'] for task in pod_statuses if task['name'] == 'node-0-server'][0]
+    wait_for_all_up_and_normal(pod_host, task_id)
 
-    # In DC/OS 1.9, task exec does not run in $MESOS_SANDBOX AND does not have access to the envvar.
-    if shakedown.dcos_version_less_than('1.10'):
-        mesos_sandbox = '/mnt/mesos/sandbox'
-    else:
-        mesos_sandbox = '$MESOS_SANDBOX'
 
-    # wait for 'nodetool status' to reflect the replacement:
-    def fun():
-        stdout = cmd.run_cli(
-            'task exec {} /bin/bash -c "cd {} && JAVA_HOME=$(ls -d jre*/) apache-cassandra-*/bin/nodetool -p 7199 status"'.format(task_id, mesos_sandbox))
-        up_ips = []
-        for line in stdout.split('\n'):
-            words = list(filter(None, line.split()))
-            if len(words) < 2:
-                continue
-            if not 'UN' == words[0]:
-                continue
-            up_ips.append(words[1])
-        utils.out('UN nodes (want {} entries without {}): {}'.format(DEFAULT_TASK_COUNT, pod_host, up_ips))
-        return len(up_ips) == DEFAULT_TASK_COUNT and not pod_host in up_ips
-    # observed to take 2-3mins in practice:
-    shakedown.wait_for(lambda: fun(), timeout_seconds=DEFAULT_CASSANDRA_TIMEOUT, sleep_seconds=15, noisy=True)
+@pytest.mark.sanity
+@utils.dcos_1_9_or_higher # dcos task exec not supported < 1.9
+def test_node_replace_replaces_seed_node():
+    pod_to_replace = 'node-0'
+    pod_host = get_pod_host(pod_to_replace)
+
+    # start replace and wait for it to finish
+    cmd.run_cli('cassandra pods replace {}'.format(pod_to_replace))
+    plan.wait_for_in_progress_recovery(PACKAGE_NAME)
```

Comment:
Can this flake if the recovery finishes or otherwise gets past IN_PROGRESS too quickly? (assuming wouldnt happen in practice but worth mentioning...)